File: GarbageV3-2025/app/src/main/java/dk/itu/GarbageV3/AddItemActivity.java
package dk.itu.GarbageV3;

import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class AddItemActivity extends AppCompatActivity {
    private static ItemsDB itemsDB;

    @Override
    protected void onCreate (Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.add_item);
        itemsDB= ItemsDB.get();

        TextView itemAddedNotification= findViewById(R.id.item_was_added_notification);
        itemAddedNotification.setVisibility(TextView.INVISIBLE);

        EditText what= findViewById(R.id.add_new_item_what_edit_text);
        EditText where= findViewById(R.id.add_new_item_where_edit_text);
        Button addItem= findViewById(R.id.add_new_item_button);

        addItem.setOnClickListener(view -> {
            itemsDB.addItem(what.getText().toString().trim(), where.getText().toString().trim());
            itemAddedNotification.setVisibility(TextView.VISIBLE);
            what.setText("");
            where.setText("");
        });

    }
}

File: GarbageV4-2025-KotlinV2/app/src/main/java/dk/itu/garbagev4_2025_kotlin/AddItemActivity.kt
package dk.itu.garbagev4_2025_kotlin

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.view.View
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity

class AddItemActivity : AppCompatActivity() {

    private lateinit var itemsDB: ItemsDB

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_add_new_item)

        itemsDB = ItemsDB.getInstance(this)

        val itemAddedNotification: TextView = findViewById(R.id.notification_item_added)
        itemAddedNotification.visibility = View.INVISIBLE

        val addNewItemButton = findViewById<Button>(R.id.add_new_item_button)

        addNewItemButton.setOnClickListener {
            val userInputWhatToSort = findViewById<TextView>(R.id.add_new_item_text).text.toString()
            val userInputWhereToSort = findViewById<TextView>(R.id.add_where_to_sort_new_item_text).text.toString()

            itemsDB.addItem(userInputWhatToSort, userInputWhereToSort)
            itemAddedNotification.visibility = View.VISIBLE
            findViewById<TextView>(R.id.add_new_item_text).text = ""
            findViewById<TextView>(R.id.add_where_to_sort_new_item_text).text = ""
            val inputMethodManager = getSystemService(INPUT_METHOD_SERVICE) as android.view.inputmethod.InputMethodManager
            inputMethodManager.hideSoftInputFromWindow(addNewItemButton.windowToken, 0)
        }

    }
}